### DISCUSSION TITLE
How to debug

### DISCUSSION THREAD

**Discussion by david-shiko** [fastapi-users/fastapi-users#583] [Q&A]
I'm sorry for a possible silly question, but how to debug/trace incoming request, where I should set breakpoint? 
I found this function and set breakpoint, but it do nothing `/fastapi_users/router/auth.py:23`

For example I'm trying to find out why user is'nt quthorized. 

(I'm successfully authorized via GUI)

```
OAuth2PasswordBearer (OAuth2, password)
Authorized

Token URL: /auth/jwt/login

Flow: password
username: user2@example.com
password: ******
Client credentials location: basic
client_secret: ****** 
```

```
@router.post("/", status_code=200)
async def create_claim(
        claim: schemas.Claim,
        user: schemas.User = Depends(fastapi_users.current_user())
):
    pass
```

Proof that user are exists

```
                  id                  |       email        |                       hashed_password                        | is_active | is_superuser | is_verified 
--------------------------------------+--------------------+--------------------------------------------------------------+-----------+--------------+-------------
 7727b673-6f33-4d07-ba0e-6f13844c2501 | user2@example.com  | $2b$12$etoknVNvAQxPzHMQfWB./eO8lWzxQeRkSHTymHYtnW/X.hOXSHobq | t         | f            | t
```

**Comment by david-shiko** [comment]
I found how it works:
You should use it as whole endpoint:
```
@app.get("/protected-route")
def protected_route(user: User = Depends(fastapi_users.current_user())):
    return f"Hello, {user.email}"
```

But here is another question, how to use this method (`current_user`) without creating additional endpoint (as shown in initial question( ?

**Reply by MatthewScholefield** [reply]
Your question seems a little unclear to me. ~~But I think your problem might be that you shouldn't call the `current_user` variable because you are passing the function itself.~~

**Reply by frankie567** [reply]
Actually, @david-shiko was right there, the correct usage since [v5.1.0](https://github.com/frankie567/fastapi-users/releases/tag/v5.1.0) is `fastapi_users.current_user()` (it's a function that returns a function, a "factory" of sort).

However, I don't get the question here:

> how to use this method (current_user) without creating additional endpoint

This method is intended to be used as a dependency to protect your routes, i.e. the routes that require a user to be authenticated.

Actually, I don't see any difference between this:

> ```py
> @router.post("/", status_code=200)
> async def create_claim(
>         claim: schemas.Claim,
>         user: schemas.User = Depends(fastapi_users.current_user())
> ):
>     pass
> ```

and this:

> ```py
> @app.get("/protected-route")
> def protected_route(user: User = Depends(fastapi_users.current_user())):
>     return f"Hello, {user.email}"
> ```

**Reply by david-shiko** [reply]
My intention is pretty simple: Get current incoming user (id or something else). 
When I renamed `/protected-route` to `foo` or any another value - it's stopped to working for me. 
What I want is to use `fastapi_users.current_user()` inside any endpoint/handler/route (i don't know difference between this words) just as I'm using my self maked function `services.get_user_by_token`

**Reply by frankie567** [reply]
Ok, I see. Well, it should work like this for any route you define, I don't see anything wrong with what you did.

Could you provide the full response / error trace you get?

**Reply by david-shiko** [reply]
Yeah, I will do it tomorrow or later, now is just deadline. BTW, what you mean 

> to protect your routes

You just have a token in headers and may appeal to any endpoint, what the problem?

**Reply by frankie567** [reply]
What I mean by "protect your routes" is simply "require the user to be authenticated on that route" 😉
